### ISSUE TITLE
[java] UnusedAssignment false positive with conditional assignments of fields

### ISSUE THREAD

**Issue by kohlschuetter** [pmd/pmd#4618]
**Affects PMD Version:** 
7.0.0-SNAPSHOT

**Rule:**

[https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#unusedassignment](UnusedAssignment)

**Description:**

**Code Sample demonstrating the issue:**

```java
Selftest.java
```

**Expected outcome:**

PMD reports violations, but that's wrong. These are false positives:
```
Selftest.java:162:	UnusedAssignment:	The value assigned to field 'withIssues' is never used (overwritten on line 168)
Selftest.java:177:	UnusedAssignment:	The value assigned to field 'withIssues' is never used (overwritten on line 184)
Selftest.java:666:	UnusedAssignment:	The value assigned to field 'modified' is never used (overwritten on lines 682 and 703)
Selftest.java:703:	UnusedAssignment:	The value assigned to field 'modified' is never used (reassigned every iteration)
Selftest.java:704:	UnusedAssignment:	The value assigned to field 'withIssues' is never used (overwritten on lines 704 and 762)
```

The values are assigned conditionally, within an if-clause.

**Running PMD through:** CLI/Maven

Files attached.
[file-and-ruleset3.zip](https://github.com/pmd/pmd/files/11909818/file-and-ruleset3.zip)

**Comment by kohlschuetter** [comment]
@oowekyala I can confirm that [34871e9](https://github.com/oowekyala/pmd/commit/34871e97e857073ae6794f0587f9cb8c5ab25777) fixes the warning for me. Cheers!
